backend/src/core/middlewares/phonebook.middleware.ts

```diff
@@ -0,0 +1,34 @@
+import { loggerWithLabel } from '@core/logger'
+import { Request, Response } from 'express'
+
+import { PhonebookService } from '@core/services/phonebook.service'
+import { ChannelType } from '@core/constants'
+
+const logger = loggerWithLabel(module)
+
+const getListsByChannel = async (
+  req: Request,
+  res: Response
+): Promise<Response | void> => {
+  logger.info({
+    message: 'Get phonebook lists by channel',
+    action: 'getListsByChannel',
+  })
+  const userId = req.session?.user?.id
+  const { channel } = req.params
+
+  try {
+    const lists = await PhonebookService.getPhonebookLists({
+      userId,
+      channel: channel as ChannelType,
+    })
+    return res.status(200).json({ lists })
+  } catch (e) {
+    // explicitly return 200 if something goes wrong with phonebook API and not an error
```

Comment:
I don't really understand — why are we returning 200 if sth goes wrong with Phonebook API

Reply:
edit: scratch that, we don't want to trigger dd alerts and we wont be bubbling the error to the user, so will retain `200` and just let user journey carry on

Reply:
I'm guessing [this](https://github.com/opengovsg/postmangovsg/pull/2042#discussion_r1229364914) is the eventual outcome right.

I agree it's better to do a 5xx error (probably 502, which is what I did for external dependencies in CheckWho) — then if it really goes down, the person on-call will be notified and we can put up banners or do other fixes